### PR TITLE
REFACTOR

### PR DESCRIPTION
**Pull request by dibashthapa**

### CODE REVIEW

File: main_application/app.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+from flask import Flask
+from views import views
```
Comment (dibashthapa) [inline main_application/app.py:2]: this `app.py` is only responsible to handle all the methods of `Flask` instance. It was fine in `main.py` file , but anyway , if we were to test this application in the future, it would have to be in different file

File: main_application/database.py [hunk 2]
```diff
@@ -1,31 +1,20 @@
-from fileinput import close
 import sqlite3 as sql
+from contextlib import contextmanager
 
-#creating the class
-class StudentDatabase :
```
Comment (dibashthapa) [inline main_application/database.py:5]: You are using classes just for the sake of using classes. The responsibilities of class could have been to fetch the students, insert their details, instead of executing dynamically passed queries and closing the connection. Instead you could have chosen `SQLite`, if you were only using it for executing dynamic queries

File: main_application/database.py [hunk 3]
```diff
@@ -1,31 +1,20 @@
...
-    def create_connection(self,database_name) :
+def connect(database=DATABASE):
+    conn = sql.connect(database)
+    conn.row_factory = sql.Row
+    return conn
 
-        self.con = sql.connect(f"{database_name}")
 
-    def execute_sql_commands(self,commands) :
+@contextmanager
```
Comment (dibashthapa) [inline main_application/database.py:13]: I have used context manager here, to remove your redundant code `db.conn.close()`. You can research more about context managers in python

File: main_application/views.py [hunk 4]
```diff
@@ -1,105 +1,59 @@
...
-        cur.execute('INSERT INTO students VALUES ( ? , ? , ? )',(new_name,new_phone_number,new_address))
-
-        con.commit()
-
-        #closing the connection
-        con.close()
-
-        
-
+        with sqlite3_connection() as db:
```
Comment (dibashthapa) [inline main_application/views.py:29]: Here you can see, you don't have to connect and close the connection each time. i do understand why you creating connection in each route, as i was getting thread id mismatch errors while globally creating connections on the top level. This was easily replaced by context managers

File: main_application/views.py [hunk 5]
```diff
@@ -1,105 +1,59 @@
...
+                (new_name, new_phone_number, new_address),
+            )
+            db.commit()
         return "Record has been updated"
 
-
 @views.route("/show-database")
-def show_database() :
-
-    #creating the connection
```
Comment (dibashthapa) [inline main_application/views.py:66]: You don't have to explicitly mention these verbose comments. These functions are self explanatory.

File: main_application/app.py (hunk 1 above)
Comment (AryanTmg628) [reply main_application/app.py:2]: You mean i need to make app.py for creating instance of FLask Instance and run the application ?

File: main_application/database.py (hunk 2 above)
Comment (AryanTmg628) [reply main_application/database.py:5]: sorry i am unable to understand this :)

File: main_application/app.py (hunk 1 above)
Comment (dibashthapa) [reply main_application/app.py:2]: > You mean i need to make app.py for creating instance of FLask Instance and run the application ?

@AryanTmg628 Yes, but run the server in another file , `app.run(debug=True)`

File: main_application/database.py (hunk 2 above)
Comment (dibashthapa) [reply main_application/database.py:5]: > sorry I am unable to understand this :)

In layman terms, you using classes here serve no purpose. It's just looks you are using classes, for the sake of learning classes, without any purpose
